### ISSUE TITLE
Can't compile quickwit on macos or any OS different from linux and android

### ISSUE THREAD

**Issue by fmassot** [quickwit-oss/quickwit#1747]
**Description**

We are using the `NetworkInterface` method `is_dormant` from the crate [pnet_datalink](https://github.com/libpnet/libpnet) but it's only available for Linux and android.

```rust
    /// Triggered when the driver has signated netif_dormant_on
    /// Check <https://www.kernel.org/doc/html/latest/networking/operstates.html> for more information
    #[cfg(any(target_os = "linux", target_os = "android"))]
    pub fn is_dormant(&self) -> bool {
        self.flags & (pnet_sys::IFF_DORMANT as InterfaceType) != 0
    }
```

https://github.com/libpnet/libpnet/blob/master/pnet_datalink/src/lib.rs#L274-L279

**Comment by fulmicoton** [comment]
@guilload can you cook a quick workaround for macos? A lot of dev use mac nowadays.

**Comment by fulmicoton** [comment]
Actually I opened a PR. @fmassot can you confirm it fixes the build?

**Comment by fmassot** [comment]
Fixed by #1765
